editor/test/editor/lua_parser_test.clj

```diff
@@ -84,6 +89,26 @@
           result (select-keys (lua-info code) [:vars :requires])]
       (is (= {:vars #{}
               :requires [[nil "mymath"]]} result))))
+  (testing "require function call with tail function call"
```

Comment:
This is the main case I fixed, but I added more tests to make sure everything is fine.